src/main/java/com/tidyjava/bp/MarkdownPost.java

```diff
@@ -0,0 +1,66 @@
+package com.tidyjava.bp;
+
+import org.commonmark.ext.front.matter.YamlFrontMatterExtension;
+import org.commonmark.ext.front.matter.YamlFrontMatterVisitor;
+import org.commonmark.html.HtmlRenderer;
+import org.commonmark.node.Node;
+import org.commonmark.parser.Parser;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.singletonList;
+
+public class MarkdownPost {
+    public static final String EXTENSION = ".md";
+
+    private Node parsedFile;
+    private Map<String, List<String>> metadata;
+    private String url;
+
+    public MarkdownPost(File file) {
```

Comment:
is constructor really a good place for it? not intuitive for sure...
What about MarkdownPost.from(file)?